x-pack/plugins/canvas/public/components/var_config/var_config.tsx

```diff
@@ -0,0 +1,227 @@
+import { DeleteVar } from './delete_var';
+
+import './var_config.scss';
+
+const { VarConfig: strings } = ComponentStrings;
+
+enum PanelMode {
+  List,
+  Edit,
+  Delete,
+}
+
+const typeToToken = {
+  number: 'tokenNumber',
+  boolean: 'tokenBoolean',
+  string: 'tokenString',
+};
+
+interface Props {
+  variables: CanvasVariable[];
+
+  onCopyVar: (v: CanvasVariable) => void;
+  onDeleteVar: (v: CanvasVariable) => void;
+  onAddVar: (v: CanvasVariable) => void;
+  onEditVar: (oldVar: CanvasVariable, newVar: CanvasVariable) => void;
+}
+
+export const VarConfig: FC<Props> = ({
+  variables,
+  onCopyVar,
+  onDeleteVar,
+  onAddVar,
+  onEditVar,
+}) => {
+  const [panelMode, setPanelMode] = useState<PanelMode>(PanelMode.List);
+  const [selectedVar, setSelectedVar] = useState<CanvasVariable | null>(null);
+
+  const selectAndEditVar = (v: CanvasVariable) => {
+    setSelectedVar(v);
+    setPanelMode(PanelMode.Edit);
+  };
+
+  const selectAndDeleteVar = (v: CanvasVariable) => {
+    setSelectedVar(v);
+    setPanelMode(PanelMode.Delete);
+  };
+
+  const actions: EuiTableActionsColumnType<CanvasVariable>['actions'] = [
+    {
+      type: 'icon',
+      name: strings.getCopyActionButtonLabel(),
+      description: strings.getCopyActionTooltipLabel(),
+      icon: 'copyClipboard',
+      onClick: onCopyVar,
+      isPrimary: true,
+    },
+    {
+      type: 'icon',
+      name: strings.getEditActionButtonLabel(),
+      description: '',
+      icon: 'pencil',
+      onClick: selectAndEditVar,
+    },
+    {
+      type: 'icon',
+      name: strings.getDeleteActionButtonLabel(),
+      description: '',
+      icon: 'trash',
+      color: 'danger',
+      onClick: selectAndDeleteVar,
+    },
+  ];
+
+  const varColumns: Array<EuiBasicTableColumn<CanvasVariable>> = [
+    {
+      field: 'type',
+      name: strings.getTableTypeLabel(),
+      sortable: true,
+      render: (varType: CanvasVariable['type'], _v: CanvasVariable) => {
+        return <EuiToken iconType={typeToToken[varType]} />;
+      },
+      width: '50px',
+    },
+    {
+      field: 'name',
+      name: strings.getTableNameLabel(),
+      sortable: true,
+    },
+    {
+      field: 'value',
+      name: strings.getTableValueLabel(),
+      sortable: true,
+      truncateText: true,
+      render: (value: CanvasVariable['value'], _v: CanvasVariable) => {
+        return '' + value;
+      },
+    },
+    {
+      actions,
+      width: '60px',
+    },
+  ];
+
+  return (
+    <div
+      className={`canvasSidebar__expandable canvasVarConfig__container ${
+        panelMode !== PanelMode.List ? 'canvasVarConfig-isEditMode' : ''
+      }`}
+    >
+      <div className="canvasVarConfig__innerContainer">
+        <EuiAccordion
+          id="accordion-variables"
+          className="canvasVarConfig__listView canvasSidebar__accordion"
+          buttonContent={
+            <EuiToolTip content={strings.getTitle()} position="left" className="canvasArg__tooltip">
```

Comment:
I noticed that this tooltip is only repeating the `Variables` title. Instead, we should either supply a more descriptive tooltip (if we feel a description is necessary for users), or omit the tooltip altogether (if we feel a description is not necessary).

Assuming we wish to give a description, I imagine something like the follow would work:
`Add variables to store and edit common values`